### PR TITLE
Improve code coverage of System.IO.FileSystem.AccessControl (35.7%)

### PR DESCRIPTION
**Pull request by 8to5Developer**
Pull request for the issue #15808

### CODE REVIEW

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+namespace System.Security.AccessControl
+{
+    public class DirectoryObjectSecurityTests
+    {
+        [Fact]
+        public void ObjectInitialization_DefaultConstructor_Success()
+        {
+            var activeDirectorySecurity = new ActiveDirectorySecurity();
+            Assert.NotNull(activeDirectorySecurity);
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: We don't need to assert that the result of a new operation is non-null. Instead we should assert that the various properties of the object match expected defaults (and if such tests exist elsewhere, this one could just be deleted).

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        }
+
+        [Fact]
+        public void ObjectInitialization_CommonSecurityDescriptor_Success()
+        {
+
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: Nit: unnecessary blank line

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+            Assert.NotNull(activeDirectorySecurity);
+        }
+
+        [Fact]
+        public void ObjectInitialization_CommonSecurityDescriptor_Success()
+        {
+            
+            CommonSecurityDescriptor descriptor = new CommonSecurityDescriptor(false, false, string.Empty);
+            var customObjectSecurity = new CustomDirectoryObjectSecurity(descriptor);
+            Assert.NotNull(customObjectSecurity);
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: Same `new` comment as earlier.  Won't comment on it further, but it applies to all other cases.

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+        }
+
+        [Fact]
+        public void ObjectInitialization_InvalidArgument()
+        {
+            Assert.Throws<ArgumentNullException>(() => new CustomDirectoryObjectSecurity(null));
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: Is there a parameter name included in the exception?  If so, it'd be good to include it as the first argument to Throws. (Same for other cases of argument exceptions)

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+            var activeDirectorySecurity = new ActiveDirectorySecurity();
+            Assert.Throws<ArgumentException>(() => 
+               activeDirectorySecurity
+               .GetAccessRules(false, false, typeof(System.Security.Principal.GenericPrincipal)));
+        }
+
+        [Fact]
+        public void GetAccessRules_TargetType_SecurityIdentifier_ReturnsValidObject()
+        {
+            CommonSecurityDescriptor descriptor = new CommonSecurityDescriptor(false, false, string.Empty);
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: Nit: CommonSecurityDescriptor => var

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+            var customObjectSecurity = new CustomDirectoryObjectSecurity(descriptor);
+
+            AuthorizationRuleCollection ruleCollection =
+                customObjectSecurity
+               .GetAccessRules(false, false, typeof(System.Security.Principal.SecurityIdentifier));
+
+            Assert.NotNull(ruleCollection);
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: Is there anything more we can validate about the rule collection beyond GetAccessRules returning a non-null instance?  Same for other occurrences of this.

File: src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs [hunk 7]
```diff
@@ -100,12 +100,28 @@ public void GetAccessControl_Filestream_InvalidArguments()
         }
 
         [Fact]
+        public void GetAccessControl_Filestream_ReturnValidObject()
+        {
+            using (var directory = new TempDirectory())
+            using (var file = new TempFile(Path.Combine(directory.Path, "file.txt")))
+            {
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs]: Nit: extra level of bracing isn't necessary

File: src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs [hunk 8]
```diff
@@ -100,12 +100,28 @@ public void GetAccessControl_Filestream_InvalidArguments()
...
+                using (FileStream fileStream = File.Open(file.Path, System.IO.FileMode.Append, System.IO.FileAccess.Write, System.IO.FileShare.None))
+                {
+                    FileSecurity fileSecurity= FileSystemAclExtensions.GetAccessControl(fileStream);
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs]: Nit: space before equals

File: src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs [hunk 9]
```diff
@@ -150,5 +166,34 @@ public void SetAccessControl_FileStream_FileSecurity_InvalidArguments()
         {
             Assert.Throws<NullReferenceException>(() => FileSystemAclExtensions.SetAccessControl((FileStream)null, (FileSecurity)null));
         }
+
+        [Fact]
+        public void SetAccessControl_FileStream_FileSecurity_InvalidFileSecurityObject()
+        {
+
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs]: Nit: unnecessary blank line

File: src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs [hunk 10]
```diff
@@ -150,5 +166,34 @@ public void SetAccessControl_FileStream_FileSecurity_InvalidArguments()
...
+            using (var directory = new TempDirectory())
+            using (var file = new TempFile(Path.Combine(directory.Path, "file.txt")))
+            {
```
Comment (stephentoub) [inline src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs]: Nit: unnecessary extra level of bracing

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs (hunk 4 above)
Comment (8to5Developer) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: Is there any document specifying the code standards for the test cases that I could go through ?

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs (hunk 2 above)
Comment (8to5Developer) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: Do you use any tool to check the Indentation?

Comment (8to5Developer) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: Do you use any specific tool to check the indentation, etc?

File: src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs (hunk 4 above)
Comment (danmosemsft) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: Not that covers that, I think. It's one of those things we just learn by getting feedback the first time..

Comment (8to5Developer) [inline src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs]: ok, thanks
I will take a look at the issues and fix them soon.

**Comment by karelz** [comment]
What is the next step? Are we waiting for code review feedback to be incorporated by @8to5Developer?

**Comment by 8to5Developer** [comment]
I just updated the files according to the suggested recommendations.

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Portable Linux x64 Debug Build

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Portable Linux x64 Release Build

File: src/System.IO.FileSystem.AccessControl/tests/FileSystemAuditRuleTests.cs [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+        [Fact]
+        public void FileSystemRights_ReturnValidObject()
+        {
+            IdentityReference identity = new SecurityIdentifier(WellKnownSidType.WorldSid, null);
+            FileSystemAuditRule auditRule = new FileSystemAuditRule(@"MYDOMAIN\MyAccount", FileSystemRights.ReadData, AuditFlags.Failure);
+            var fileSystemRights= auditRule.FileSystemRights;
+            Assert.NotNull(fileSystemRights);
+        }
+
```
Comment (ianhays) [inline src/System.IO.FileSystem.AccessControl/tests/FileSystemAuditRuleTests.cs]: nit: you've got a bunch of extra newlines scattered about.

File: src/System.IO.FileSystem.AccessControl/tests/FileSystemAuditRuleTests.cs [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+            FileSystemAuditRule auditRule = new FileSystemAuditRule(@"MYDOMAIN\MyAccount", FileSystemRights.ReadData, AuditFlags.Failure);
+            Assert.NotNull(auditRule);
+        }
+
+        [Fact]
+        public void ObjectInitialization_InvalidFileSystemRights()
+        {
+            FileSystemRights fileSystemRights = (FileSystemRights)(-1);
+            IdentityReference identity = new SecurityIdentifier(WellKnownSidType.WorldSid, null);
+            Assert.Throws< ArgumentOutOfRangeException>(()=> new FileSystemAuditRule(@"MYDOMAIN\MyAccount", fileSystemRights, AuditFlags.Failure));
```
Comment (ianhays) [inline src/System.IO.FileSystem.AccessControl/tests/FileSystemAuditRuleTests.cs]: nit: extra space before the AOoRE.

tip: we've got a tool that will fix the formatting for you: https://github.com/dotnet/codeformatter

**Pull request merged**